### PR TITLE
Prevent pushbuttons with 'blocked' controls getting stuck pressed

### PR DESCRIPTION
**Pull request by ronso0**
When a disallowed click is performed, like enabling Passthrough, VinylControl or TalkOver with no respective device set up, the button release is not registered because modal message boxes block interaction with Mixxx main window, and the button is 'pressed' only after the message is closed and remains so. Noticed with Shade (colored button) and LateNight (3D effect).
* click Passthrough button
* `group,passthrough` is set `1`
* message pops up, blocks release event, `group,passthrough` is forced `0`
* close popup
* button is set 'pressed', no further control update

Solution:
* set the 'pressed' property before setting any control (which could enforce a popup)
* allow `onConnectedControlChanged` to update the 'pressed' property also for pushbutton with more than one state 

I didn't discover a regression, yet, so I'm curious why this limitation was installed in the first place.?

Maybe we could also use a special pushbutton widget or a different control connection for those buttons that checks if setting the control is allowed to be toggled **before** actually doing it, which would also prevent disallowed Passthrough stopping playing decks.
But IMO the fix proposed here is the easiest way to resolve this issue.

### CODE REVIEW

File: src/widget/wpushbutton.cpp [hunk 1]
```diff
@@ -290,9 +290,7 @@ void WPushButton::onConnectedControlChanged(double dParameter, double dValue) {
     Q_UNUSED(dParameter);
     // Enums are not currently represented using parameter space so it doesn't
     // make sense to use the parameter here yet.
-    if (m_iNoStates == 1) {
-        m_bPressed = (dValue == 1.0);
-    }
+    m_bPressed = (dValue == 1.0);
```
Comment (ronso0) [inline src/widget/wpushbutton.cpp]: >= 1 ?

File: src/widget/wpushbutton.cpp [hunk 2]
```diff
@@ -290,9 +290,7 @@ void WPushButton::onConnectedControlChanged(double dParameter, double dValue) {
     Q_UNUSED(dParameter);
     // Enums are not currently represented using parameter space so it doesn't
     // make sense to use the parameter here yet.
-    if (m_iNoStates == 1) {
```
Comment (daschuer) [inline src/widget/wpushbutton.cpp]: This code is only a workaround for one state buttons where the pressed state matches the value.

**Review by daschuer — CHANGES_REQUESTED** [review]
This has the issue, when I press the play button and hold it until the track is at end, the m_bPressed state is lost. This does not happen with the master build

**Comment by daschuer** [comment]
It looks like this is a known bug: 
https://bugreports.qt.io/browse/QTBUG-50051

**Comment by daschuer** [comment]
I have found a alternative solution here: 
https://github.com/ronso0/mixxx/pull/6

**Comment by daschuer** [comment]
Can one confirm that this is working on Windows as well?
I am afraid to ask for MacOs ..:-/ to not block this PR.

**Comment by daschuer** [comment]
Ci issues are unrelated.

**Comment by daschuer** [comment]
Let's merge this. We can test it during alpha and beta phase on the other OSs.

**Pull request merged**